src/block.rs

```diff
@@ -278,3 +275,40 @@ impl<B: BlockDevice> BlockDevice for CachedBlockDevice<B> {
         self.block_device.count()
     }
 }
+
+#[cfg(feature = "std")]
+impl BlockDevice for std::fs::File {
```

Comment:
Might want to implement it on &std::fs::File. Read is implemented on &std::fs::File.

Reply:
Hum, no it's not. Besides, it takes &mut as self, so it's easier to implement it on File than on &mut File